Cpp_test/lib/ipc-pipe-send.cpp

```diff
@@ -0,0 +1,76 @@
+#include "ipc-pipe-send.h"
+
+IPCPipeSend::~IPCPipeSend()
+{
+	close(this->pd);
+	unlink(this->info.method_name);
+}
+
+void IPCPipeSend::init()
+{
+	this->file_handler.setup_file(info.file_name, FileMode::READ);
+	// remove old pipe (if any)
+	bool remove_old_pipe = remove(this->info.method_name);
```

Comment:
remove() call does not return a bool. 

But this won't work, if there is a process which has the pipe opened remove() will still return 0 (as a success if we read the documentation).

I tried with this code to confirm:
```c++
// Example program
#include <iostream>
#include <string>
#include <vector>
#include <thread>
#include <sys/types.h>
#include <sys/stat.h>
#include <fcntl.h>
#include <unistd.h>


int main()
{
    mkfifo("myFifo", 0660);
    
    std::thread reader_thread([](){
        std::cout<<"Opening the pipe"<< std::endl;
        int fd = open("myFifo", O_RDWR | O_NONBLOCK);
        sleep(5);
        std::cout<<"I am still in the pipe"<< std::endl;
        close(fd);
    });
    sleep(1);
    std::thread sender_thread([](){
        std::cout<<"Removing the pipe"<< std::endl;
        int try_removing = remove("myFifo");
        std::cout << "status of try_removing: " << try_removing << std::endl;
    });
    
    
    
    reader_thread.join();
    sender_thread.join();
}
```
and it returns:
```
Opening the pipe
Removing the pipe
status of try_removing: 0
I am still in the pipe
```

Reply:
Yes, you are right. There is also a remove function bool remove(const path& p), but this int remove(const char *pathname); is used here. With this, I want to remove the old pipe but if the new pipe is created so the old pipe cannot be removed, it will be marked to be remove after the new pipe process is done.